src/main/java/com/hcl/appscan/jenkins/plugin/scanners/StaticAnalyzer.java

```diff
@@ -28,31 +28,40 @@ public class StaticAnalyzer extends Scanner {
 	private static final String STATIC_ANALYZER = "Static Analyzer"; //$NON-NLS-1$
         
         private boolean m_openSourceOnly;
+        private String m_includeSCAGenerateIRX;
+        private boolean m_hasOptionsUploadDirect;
+        private boolean m_includeSCAUploadDirect;
         private boolean m_sourceCodeOnly;
         private String m_scanMethod;
         private String m_scanSpeed;
         
         @Deprecated
         public StaticAnalyzer(String target){
-            this(target,false);
+            this(target,true, false, false, EMPTY, EMPTY, EMPTY, false, false);
         }
         
-        public StaticAnalyzer(String target, boolean hasOptions, boolean openSourceOnly, boolean sourceCodeOnly, String scanMethod, String scanSpeed){
+        public StaticAnalyzer(String target, boolean hasOptions, boolean openSourceOnly, boolean sourceCodeOnly, String scanMethod, String scanSpeed, String includeSCAGenerateIRX, boolean hasOptionsUploadDirect, boolean includeSCAUploadDirect){
             super(target, hasOptions);
-            m_openSourceOnly=openSourceOnly;
+            m_openSourceOnly = openSourceOnly;
             m_sourceCodeOnly=sourceCodeOnly;
             m_scanMethod= scanMethod;
             m_scanSpeed=scanSpeed;
+            m_includeSCAGenerateIRX=includeSCAGenerateIRX;
+            m_hasOptionsUploadDirect=hasOptionsUploadDirect;
+            m_includeSCAUploadDirect=includeSCAUploadDirect;
         }
         
 	@DataBoundConstructor
-	public StaticAnalyzer(String target,boolean hasOptions) {
+	public StaticAnalyzer(String target,boolean hasOptions, boolean hasOptionsUploadDirect) {
 		super(target, hasOptions);
                 m_openSourceOnly=false;
                 m_sourceCodeOnly=false;
                 m_scanMethod=CoreConstants.CREATE_IRX;
                 m_scanSpeed="";
-	}
+                m_includeSCAGenerateIRX="true";
```

Comment:
I think it is better to have two different names for the these checkbox as the position of these checks are different in jelly file, it will hamper the value of the checkbox if one of the check is false.